services/shhext/envelopes.go

```diff
@@ -193,7 +194,30 @@ func (m *EnvelopesMonitor) handleAcknowledgedBatch(event whisper.EnvelopeEvent)
 		log.Debug("batch is not found", "batch", event.Batch)
 	}
 	log.Debug("received a confirmation", "batch", event.Batch, "peer", event.Peer)
+	envelopeErrors, ok := event.Data.([]whisper.EnvelopeError)
+	if !ok {
+		log.Error("received unexpected data in the the confirmation event", "batch", event.Batch)
+	}
+	failedEnvelopes := map[common.Hash]struct{}{}
+	for i := range envelopeErrors {
```

Comment:
Why not 

```suggestion
	for i, envelopeError := range envelopeErrors {
```

Reply:
to avoid copying object. EnvelopeError is not a pointer, so full object will be copied if accessed this way

Reply:
gotcha